internal/server/api.go

```diff
@@ -19,64 +19,79 @@ package server
 import (
 	"context"
 	"fmt"
-	"net/http"
 	"strings"
 
 	"github.com/thestormforge/optimize-controller/v2/internal/version"
 	"github.com/thestormforge/optimize-go/pkg/api"
+	applications "github.com/thestormforge/optimize-go/pkg/api/applications/v2"
 	experimentsv1alpha1 "github.com/thestormforge/optimize-go/pkg/api/experiments/v1alpha1"
 	"github.com/thestormforge/optimize-go/pkg/config"
 )
 
 const audience = "https://api.carbonrelay.io/v1/"
 
 func NewExperimentAPI(ctx context.Context, uaComment string) (experimentsv1alpha1.API, error) {
-	address, rt, err := loadConfig(ctx, uaComment)
+	client, err := newClientFromConfig(ctx, uaComment)
 	if err != nil {
 		return nil, err
 	}
 
-	// Create a new Experiment API client
-	c, err := api.NewClient(address, rt)
+	expAPI := experimentsv1alpha1.NewAPI(client)
+
+	// An unauthorized error means we will never be able to connect without changing the credentials and restarting
+	if _, err := expAPI.CheckEndpoint(ctx); api.IsUnauthorized(err) {
+		return nil, fmt.Errorf("experiments API is unavailable, skipping setup: %w", err)
+	}
+
+	return expAPI, nil
+}
+
+func NewApplicationAPI(ctx context.Context, uaComment string) (applications.API, error) {
+	client, err := newClientFromConfig(ctx, uaComment)
 	if err != nil {
 		return nil, err
 	}
 
-	expAPI := experimentsv1alpha1.NewAPI(c)
+	appAPI := applications.NewAPI(client)
 
-	// An unauthorized error means we will never be able to connect without changing the credentials and restarting
-	if _, err := expAPI.Options(ctx); api.IsUnauthorized(err) {
-		return nil, fmt.Errorf("experiments API is unavailable, skipping setup: %s", err.Error())
+	if _, err := appAPI.CheckEndpoint(ctx); api.IsUnauthorized(err) {
+		return nil, fmt.Errorf("applications API is unavailable, skipping setup: %w", err)
 	}
 
-	return expAPI, nil
+	return appAPI, nil
 }
 
-func loadConfig(ctx context.Context, uaComment string) (string, http.RoundTripper, error) {
+func newClientFromConfig(ctx context.Context, uaComment string) (api.Client, error) {
 	// Load the configuration
 	cfg := &config.OptimizeConfig{}
 	cfg.AuthorizationParameters = map[string][]string{
 		"audience": {audience},
 	}
 
 	if err := cfg.Load(); err != nil {
-		return "", nil, err
+		return nil, err
 	}
 
 	// Get the Experiments API endpoint from the configuration
 	// NOTE: The current version of the configuration has an explicit configuration for the
 	// experiments endpoint which would duplicate the "/experiments/" path segment
 	srv, err := config.CurrentServer(cfg.Reader())
 	if err != nil {
-		return "", nil, err
+		return nil, err
 	}
 
 	address := strings.TrimSuffix(srv.API.ExperimentsEndpoint, "/v1/experiments/")
```

Comment:
This is a dumb problem to have, but the configuration model and the API client model diverged when the API version became part of the service path. The unfortunate side effect is that:
```
address := strings.TrimSuffix(srv.API.ExperimentsEndpoint, "/v1/experiments/")
```
Needs to be:
```
address := strings.TrimSuffix(srv.API.ApplicationsEndpoint, "/v2/applications/")
```
Even though they _should_ effectively produce the same value.